### PR TITLE
Process DeleteSnapshot CRs without Phase

### PR DESCRIPTION
**Pull request by deepakkinni**
**What this PR does / why we need it**:
Users creating a DeleteSnapshot CR will be unable to set the `Status` as it's a subresource. In the current implementation, we process DeleteSnapshot only if it's in a `New` state. This change processes DeleteSnapshot CR without a set phase to `New`, this triggers the usual workflow.

This can be used to delete any fcd snapshots that were left over due to a failed upload.

**Testing**
Created a DeleteSnapshot CR and it was processed by the backup-driver
```
dkinni@dkinni-a02 vanilla-testing % cat delete-snap-1.yaml
apiVersion: backupdriver.cnsdp.vmware.com/v1alpha1
kind: DeleteSnapshot
metadata:
  name: delete-snap-3
  namespace: velero
status:
  phase: New
spec:
  backupRepository: br-1fdcebc9-6080-4eb4-8b94-9dbceb7e098b
  snapshotID: pvc:demo-app/nginx-logs:aXZkOjk4YTViYTU0LTBiMzgtNDBmNi1iYTk1LTMxNjJmMTkxYTdiYTo4MGQwYTRhMC0xMjNmLTQ3NmYtYmFlOC00N2FmOGY0NzZhYzk
```
backup-driver logs:
```
ime="2022-02-03T04:58:56Z" level=info msg="No Phase set for the DeleteSnapshot CR: velero/delete-snap-3, will explicitly update to New" controller=BackupDriverController logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/backupdriver/backup_driver_controller_base.go:802"
time="2022-02-03T04:58:56Z" level=info msg="updateDeleteSnapshotStatusPhase: DeleteSnapshot velero/delete-snap-3 updated phase from  to New" controller=BackupDriverController logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/backupdriver/backup_driver_controller.go:414"
time="2022-02-03T04:58:56Z" level=info msg="syncDeleteSnapshotByKey: calling deleteSnapshot velero/delete-snap-3 " controller=BackupDriverController logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/backupdriver/backup_driver_controller_base.go:834"
time="2022-02-03T04:58:56Z" level=info msg="deleteSnapshot called with SnapshotID pvc:demo-app/nginx-logs:aXZkOjk4YTViYTU0LTBiMzgtNDBmNi1iYTk1LTMxNjJmMTkxYTdiYTo4MGQwYTRhMC0xMjNmLTQ3NmYtYmFlOC00N2FmOGY0NzZhYzk, Namespace: velero, Name: delete-snap-3" controller=BackupDriverController logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/backupdriver/backup_driver_controller.go:168"
time="2022-02-03T04:58:56Z" level=info msg="Calling Snapshot Manager to delete snapshot with snapshotID pvc:demo-app/nginx-logs:aXZkOjk4YTViYTU0LTBiMzgtNDBmNi1iYTk1LTMxNjJmMTkxYTdiYTo4MGQwYTRhMC0xMjNmLTQ3NmYtYmFlOC00N2FmOGY0NzZhYzk" controller=BackupDriverController logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/backupdriver/backup_driver_controller.go:173"
2022/02/03 04:58:56 pei = pvc:demo-app/nginx-logs:aXZkOjk4YTViYTU0LTBiMzgtNDBmNi1iYTk1LTMxNjJmMTkxYTdiYTo4MGQwYTRhMC0xMjNmLTQ3NmYtYmFlOC00N2FmOGY0NzZhYzk
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Process DeleteSnapshot CRs without explicit Phase set
```
Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/667/

File: pkg/backupdriver/backup_driver_controller_base.go [hunk 1]
```diff
@@ -798,6 +798,19 @@ func (ctrl *backupDriverController) syncDeleteSnapshotByKey(key string) error {
 		return err
 	}
 
+	if delSnapshot.Status.Phase == "" {
+		ctrl.logger.Infof("No Phase set for the DeleteSnapshot CR: %s/%s, will be explicitly update to New", namespace, name)
```
Comment (xing-yang) [inline pkg/backupdriver/backup_driver_controller_base.go:802]: nit: `will update to New`

Can you add a comment to explain why it needs to be updated to "New"?

File: pkg/backupdriver/backup_driver_controller_base.go [hunk 2]
```diff
@@ -798,6 +798,19 @@ func (ctrl *backupDriverController) syncDeleteSnapshotByKey(key string) error {
...
+		var statusUpdateErr error
+		deleteSnapshotStatusFields := make(map[string]interface{})
+		deleteSnapshotStatusFields["Message"] = "Explicit update to New Phase"
+		delSnapshot, statusUpdateErr = ctrl.updateDeleteSnapshotStatusPhase(context.TODO(), delSnapshot.Namespace, delSnapshot.Name,
+			backupdriverapi.DeleteSnapshotPhaseNew, deleteSnapshotStatusFields)
+		if statusUpdateErr != nil {
+			ctrl.logger.Errorf("Failed to update an unset Phase of DeleteSnapshot %s/%s failed: %v", namespace, name, err)
```
Comment (xing-yang) [inline pkg/backupdriver/backup_driver_controller_base.go:809]: Remove "failed" at the end.  It's already mentioned in the beginning.
Mention you try to set to "New"?

**Review by lintongj — APPROVED** [review]
Looks good. Please be sure to resolve Xing's comments

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/672/

**Review by xing-yang — APPROVED** [review]

**Comment by xing-yang** [comment]
LGTM

**Pull request merged**